pkg/sign/certificate.go

```diff
@@ -133,17 +135,31 @@ func (f *Fulcio) GetCertificate(keypair Keypair, identityToken string) ([]byte,
 		client.Timeout = f.options.Timeout
 	}
 
-	request, err := http.NewRequest("POST", f.options.BaseURL+"/api/v2/signingCert", requestBytes)
-	if err != nil {
-		return nil, err
-	}
-	request.Header.Add("Authorization", "Bearer "+identityToken)
-	request.Header.Add("Content-Type", "application/json")
-	request.Header.Add("User-Agent", constructUserAgent(f.options.LibraryVersion))
-
-	response, err := client.Do(request)
-	if err != nil {
-		return nil, err
+	attempts := uint(0)
+	var response *http.Response
+
+	for attempts <= f.options.Retries {
+		request, err := http.NewRequest("POST", f.options.BaseURL+"/api/v2/signingCert", bytes.NewBuffer(requestJSON))
+		if err != nil {
+			return nil, err
+		}
+		request.Header.Add("Authorization", "Bearer "+identityToken)
+		request.Header.Add("Content-Type", "application/json")
+		request.Header.Add("User-Agent", constructUserAgent(f.options.LibraryVersion))
+
+		response, err = client.Do(request)
+		if err != nil {
+			return nil, err
+		}
+
+		if !(response.StatusCode >= 500 && response.StatusCode < 600) {
+			// Not a HTTP 5XX error, don't retry
+			break
+		}
+
+		attempts++
+		delay := time.Duration(math.Pow(2, float64(attempts)))
```

Comment:
Small thing, I would consider starting with 2^0=1 second, so attempts - 1, otherwise 2 seconds adds a lot of delay initially.

FWIW, I'd consider using https://github.com/hashicorp/go-retryablehttp which implements this already. This also makes the backoff function configurable.

Reply:
https://github.com/cenkalti/backoff is another library I've used for this type of thing.

I'd also add context to these calls.

Reply:
Oh, I also bumped `attempts++` down so the first sleep is shorter.